restructuring.py

```diff
@@ -0,0 +1,404 @@
+
+
+    class Url_Block:
+
+        last_run = 21092
+        url_suffix = "/block?height="
+
+        def __init__ (self, url, block_height):
+            self.url = url 
+            self.block_height = block_height 
+
+
+        def get_block_urls(self):
+
+            if self.last_run <= self.block_height:
+                start = self.last_run 
+            else:
+                start = self.block_height
+            
+            end = self.block_height + 1
+
+            url_block = []
+            for number in range (start, end):
+                block = ("http://"+ i + self.url_suffix + str(number))
+                url_block.append(block)
+
+            return url_block    
+
+    url_block = Url_Block(i, status.get_block_height())
+    # url_block.get_block_urls()
+
+
+    class Url_Validators(Url_Block):
+
+        url_suffix = "/validators?height="
+
+        def get_validators_urls(self):
+
+            if self.last_run <= self.block_height:
+                start = self.last_run 
+            else:
+                start = self.block_height
+            
+            end = self.block_height + 1
+
+            url_validators = []
+            for number in range (start, end):
+                validators = ("http://"+ i + self.url_suffix + str(number))
+                url_validators.append(validators)
+
+            return url_validators     
+
+    url_validators = Url_Validators(i, status.get_block_height())
+    # url_validators.get_validators_urls()
+            
+
+    class Blockcount:
+
+        def __init__(self, url, url_block):
+            self.url = url
+            self.url_block = url_block 
+
+
+        # get validators at block height
+        def get_blockcount(self):
+            print colored ("Scanning all blocks...", 'green')
+
+            #create classes
+            class BlockHeight():
+                def __init__(self, json):
+                    self.blockheight=json["result"]["block"]
+
+            total_blocks = len(self.url_block)
+            # print total_blocks 
+            blockcount = dict()
+            for i in self.url_block:
+                url_data = json.load(urllib2.urlopen(i))
+                block = BlockHeight (url_data)
+                # print("Got " + i)
+                for k in block.blockheight["last_commit"]["precommits"]:
+                        try:
+                            if k['validator_address']:
+                                if k['validator_address'] not in blockcount:
+                                    blockcount[k['validator_address']] = 1
+                                else:
+                                    blockcount[k['validator_address']] += 1
+                        except:
+                            pass
+
+            self.participation(blockcount)
+            return blockcount 
+
+        # get validators at block height
+        def get_block_validators(self):
+            #create classes
+            class BlockHeight():
```

Comment:
Why not put `BlockHeight` as a top level class so you don't have to duplicate code?